src/main/java/hudson/security/GlobalMatrixAuthorizationStrategy.java

```diff
@@ -473,12 +473,16 @@ public FormValidation doCheckName_(@Nonnull String value, @Nonnull AccessControl
 
             if(v.equals("authenticated"))
                 // system reserved group
-                return FormValidation.respond(Kind.OK, makeImg("user.png", "Group", false) +ev);
+                return FormValidation.respond(Kind.OK, makeImg("user.png", ev, "Group", false));
 
             try {
                 try {
                     sr.loadUserByUsername(v);
-                    return FormValidation.respond(Kind.OK, makeImg("person.png", "User", false)+ev);
+                    User u = User.get(v);
```

Comment:
This call will make Jenkins to create a user even if a bogus security realm didn't load/report it properly. It is not a big deal, but I would recommend to get user without creation and use `if (u == null || ev.equals(u.getFullName()))` in the next clause. 🐜 .

Reply:
It's an ephemeral user though. And only ever created if it's configured (or about to be configured) in the security realm. Doesn't seem like a big deal to me.